pallets/pools/src/tranche.rs

```diff
@@ -1455,12 +1483,252 @@ where
 
 #[cfg(test)]
 pub mod test {
-	#[test]
-	fn reverse_slice_panics_on_out_of_bounds() {}
+	use super::*;
+	use common_types::{PoolId, TrancheId};
+
+	// NOTE: We currently expose types in runtime-common. As we do not want
+	//       this dependecy in our pallets, we generate the types manually here.
+	//       Not sure, if we should rather allow dev-dependency to runtime-common.
+	type Balance = u128;
+	type Rate = sp_arithmetic::FixedU128;
+	type Weight = u128;
+	type TTrancheType = TrancheType<Rate>;
+	type TTranche = Tranche<Balance, Rate, Weight, CurrencyId>;
+	type TTranches = Tranches<Balance, Rate, Weight, CurrencyId, TrancheId, PoolId>;
+
+	const ONE_IN_CURRENCY: Balance = 1_000_000_000_000u128;
+	const SECS_PER_YEAR: u64 = 365 * 24 * 60 * 60;
+	const DEFAULT_POOL_ID: PoolId = 0;
+	const DEFAULT_TIME_NOW: Moment = 0;
+
+	struct TrancheTokenImpl;
+
+	impl TrancheTokenT<PoolId, TrancheId, CurrencyId> for TrancheTokenImpl {
+		fn tranche_token(pool: PoolId, tranche: TrancheId) -> CurrencyId {
+			CurrencyId::Tranche(pool, tranche)
+		}
+	}
+
+	fn residual(id: u8) -> TTranche {
+		TTranche {
+			tranche_type: TrancheType::Residual,
+			seniority: 0,
+			currency: CurrencyId::Tranche(DEFAULT_POOL_ID, [id; 16]),
```

Comment:
This one here must be `TrancheCurrency::generate(DEFUALT_POOL_ID, [id; 16])`

Reply:
I'm currently not finding a definition or implementation for `TrancheCurrency`, is there something else that should be used, ,am I missing it, or should I add this?